### PR TITLE
Add more tests for functions updating the tasks

### PR DESCRIPTION
**Pull request by SFarida**
### This PR contains the following
- a jest-localstorage-mock installed and configured in setups/localstorage.js
- a new file to implement tests app.test.js
- app.test.js contains test for the addTask() and removeTask(), editTask(), completed(), clearCompleted() and removeTask().

### CODE REVIEW

File: src/app.test.js [hunk 1]
```diff
@@ -1,10 +1,27 @@
...
     expect(addTask('sleep')).toEqual([{ completed: false, description: 'sleep', index: 1 }]);
   });
+  test('Add task "cook"', () => {
+    expect(addTask('cook')).toEqual([{ completed: false, description: 'sleep', index: 1 }, { completed: false, description: 'cook', index: 2 }]);
+  });
+  test('Edit task "sleep"', () => {
+    editTask(1, 'sleeping');
+    expect(JSON.parse(localStorage.getItem('tasks'))).toEqual([{ completed: false, description: 'sleeping', index: 1 }, { completed: false, description: 'cook', index: 2 }]);
+  });
+  test('Completed task "sleeping"', () => {
```
Comment (t-yanick) [inline src/app.test.js:17]: - [ ] Great job adding the pure fuctions test. As per project requirements, your are to use [mock](  https://robertmarshall.dev/blog/how-to-mock-local-storage-in-jest-tests/ ) for functions that use `localstorage`, `DOM` and `external APIs`. But I cannot find the use of mocks anywhere.  Kindly add this.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @SFarida , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file and PR

There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/app.test.js (hunk 1 above)
Comment (SFarida) [reply src/app.test.js:17]: hello @t-yanick thank you for your review, in the file `__setups__/localstorage.js` you'll see the necessary imports. In `package.json` on line 44 you will see the name of the package used `jest-localstorage-mock`. 
Also in `jest.config.js` you can see it there. 
Including it in the `__setups__/localstorage.js` file made it global and available to use in `app.test.js` like I did. I hope you can consider this, thanks again

Comment (t-yanick) [reply src/app.test.js:17]: Thanks for the reply. Kindly submit for re-review and the next reviewer will approve the PR.

Comment (SFarida) [reply src/app.test.js:17]: Thank you

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**